### ISSUE TITLE
CosmosDB - State Component - StateTransactionRequest with etag crashes

### ISSUE THREAD

**Issue by perokvist** [dapr/components-contrib#2319]
## Expected Behavior

Using the .NET SDK when creating a StateTransactionRequest, an etag could be passed.
```
 new StateTransactionRequest(
                    "test", JsonSerializer.SerializeToUtf8Bytes("testing"), 
                    StateOperationType.Upsert, etag: "test")
```

## Actual Behavior

`client.ExecuteStateTransactionAsync("statestore", requests, null)`
When setting any etag, dapr exits with code 2, and the following error is logged.

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x3f1fc6c]

goroutine 157 [running]:
github.com/dapr/components-contrib/state/azure/cosmosdb.(*StateStore).Multi(0xc0008d0630, 0xc0009b33a0)
        C:/Users/runneradmin/go/pkg/mod/github.com/dapr/components-contrib@v1.9.4/state/azure/cosmosdb/cosmosdb.go:375 +0x90c
```

## Steps to Reproduce the Problem
Version 1.9.4
Dapr.Client 1.9

- Create a StateTransactionRequest with an etag
- client.ExecuteStateTransactionAsync("statestore", requests, null)
- Crash

## Release Note

RELEASE NOTE:

**Comment by ItalyPaleAle** [comment]
@RyanLettieri can you please take a look?

**Comment by RyanLettieri** [comment]
@ItalyPaleAle It looks like the issue might be with the way we declare "options" inside the "Multi" function. After making the following change, it seems to work locally:
`var options *azcosmos.TransactionalBatchItemOptions` to `options := &azcosmos.TransactionalBatchItemOptions{}`

I'm running a few tests now and adding in the test case to avoid this from happening again.

**Comment by ItalyPaleAle** [comment]
Fixed by #2322
